translib/transformer/xfmr_intf.go

```diff
@@ -0,0 +1,2143 @@
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/openconfig-if-ip:ipv6/addresses/address/state") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv6/addresses/address/state") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv6/state") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/openconfig-if-ip:ipv6/state") {
+		tblList = append(tblList, intTbl.appDb.intfTN)
+	} else if strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv4/addresses") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/openconfig-if-ip:ipv4/addresses") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/openconfig-if-ip:ipv6/addresses") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv6/addresses") {
+		tblList = append(tblList, intTbl.cfgDb.intfTN)
+	} else if inParams.oper == GET && strings.HasPrefix(targetUriXpath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv4/neighbors") ||
+		strings.HasPrefix(targetUriXpath, "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv6/neighbors") {
+		tblList = append(tblList, "NONE")
+	} else if strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/ethernet") ||
+		strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface/openconfig-if-ethernet:ethernet") {
+		if inParams.oper != DELETE {
+			tblList = append(tblList, intTbl.cfgDb.portTN)
+		}
+	} else if targetUriPath == "/openconfig-interfaces:interfaces/interface" {
+		tblList = append(tblList, intTbl.cfgDb.portTN)
+	} else if strings.HasPrefix(targetUriPath, "/openconfig-interfaces:interfaces/interface") {
+		if inParams.oper != DELETE {
+			tblList = append(tblList, intTbl.cfgDb.portTN)
+		}
+	} else {
+		err = errors.New("Invalid URI")
+	}
+
+	if log.V(3) {
+		log.Infof("TableXfmrFunc - Uri: (%v), targetUriPath: %s, tblList: (%v)", inParams.uri, targetUriPath, tblList)
+	}
+
+	return tblList, err
+}
+
+var YangToDb_intf_tbl_key_xfmr KeyXfmrYangToDb = func(inParams XfmrParams) (string, error) {
+	var err error
+
+	pathInfo := NewPathInfo(inParams.uri)
+	reqpathInfo := NewPathInfo(inParams.requestUri)
+	requestUriPath := reqpathInfo.YangPath
+
+	log.Infof("YangToDb_intf_tbl_key_xfmr: inParams.uri: %s, pathInfo: %s, inParams.requestUri: %s", inParams.uri, pathInfo, requestUriPath)
+
+	ifName := pathInfo.Var("name")
+	idx := reqpathInfo.Var("index")
+	var i32 uint32
+	i32 = 0
+
+	if idx != "" {
+		i64, _ := strconv.ParseUint(idx, 10, 32)
+		i32 = uint32(i64)
```

Comment:
maybe we should return error if i32 != 0, because subinterface is not supported

Reply:
@sayedsaquibbrcm I've addressed the review comments.